lib/signaling/v2/peerconnectionmanager.js

```diff
@@ -313,27 +317,34 @@ function createDummyAudioMediaStreamTrack(audioContext) {
  */
 
 /**
- * Apply {@link MediaStreamTrackAndDataTrackSenderChanges}.
+ * Apply {@link TrackSenderChanges}.
  * @param {PeerConnectionManager} peerConnectionManager
- * @param {MediaStreamTrackAndDataTrackSenderChanges} changes
+ * @param {TrackSenderChanges} changes
  * @returns {void}
  */
-function applyMediaStreamTrackAndDataTrackSenderChanges(peerConnectionManager, changes) {
+function applyTrackSenderChanges(peerConnectionManager, changes) {
   if (changes.data.add.size
     || changes.data.remove.size
     || changes.media.add.size
     || changes.media.remove.size) {
     peerConnectionManager._peerConnections.forEach(function(peerConnection) {
-      peerConnection.removeMediaStream(this._localMediaStream);
       changes.data.remove.forEach(peerConnection.removeDataTrackSender, peerConnection);
+      changes.media.remove.forEach(peerConnection.removeMediaTrackSender, peerConnection);
+    });
+
+    changes.media.remove.forEach(function(mediaTrackSender) {
+      this._localMediaStream.removeTrack(mediaTrackSender.track);
```

Comment:
I would really like us to be able to get rid of this. If the logic for handling the MediaStream could move into PeerConnectionV2, would this simplify things?

I know you had some hesitation on this, due to our bug with too many MediaStreams, but the Mozilla team confirmed it was because we ended up created tens of thousands or more... 1 MediaStream per PeerConnectionV2, esp. if we clean them up, should be safe, IMO.